cblox/include/cblox/core/submap_collection_inl.h

```diff
@@ -0,0 +1,376 @@
+  CHECK(it != id_to_submap_.end());
+  return *it->second;
+}
+
+template <typename SubmapType>
+const std::vector<typename SubmapType::Ptr>
+SubmapCollection<SubmapType>::getSubMaps() const {
+  std::vector<typename SubmapType::Ptr> submap_ptrs;
+  for (const auto& id_submap_pair : id_to_submap_) {
+    submap_ptrs.emplace_back(id_submap_pair.second);
+  }
+  return submap_ptrs;
+}
+
+// Gets a pointer to the tsdf_map member of the active submap
+template <typename SubmapType>
+TsdfMap::Ptr SubmapCollection<SubmapType>::getActiveTsdfMapPtr() {
+  const auto it = id_to_submap_.find(active_submap_id_);
+  CHECK(it != id_to_submap_.end());
+  return (it->second)->getTsdfMapPtr();
+}
+// Gets a reference to tsdf_map member of the active submap
+template <typename SubmapType>
+const TsdfMap& SubmapCollection<SubmapType>::getActiveTsdfMap() const {
+  const auto it = id_to_submap_.find(active_submap_id_);
+  CHECK(it != id_to_submap_.end());
+  return (it->second)->getTsdfMap();
+}
+
+// Gets a reference to the active submap
+template <typename SubmapType>
+const SubmapType& SubmapCollection<SubmapType>::getActiveSubMap() const {
+  const auto it = id_to_submap_.find(active_submap_id_);
+  CHECK(it != id_to_submap_.end());
+  return *(it->second);
+}
+
+template <typename SubmapType>
+const Transformation& SubmapCollection<SubmapType>::getActiveSubMapPose()
+    const {
+  return getActiveSubMap().getPose();
+}
+template <typename SubmapType>
+const SubmapID SubmapCollection<SubmapType>::getActiveSubMapID() const {
+  return active_submap_id_;
+}
+
+template <typename SubmapType>
+TsdfMap::Ptr SubmapCollection<SubmapType>::getProjectedMap() const {
+  // Creating the global tsdf map and getting its tsdf layer
+  TsdfMap::Ptr projected_tsdf_map_ptr =
+      TsdfMap::Ptr(new TsdfMap(submap_config_));
+  Layer<TsdfVoxel>* combined_tsdf_layer_ptr =
+      projected_tsdf_map_ptr->getTsdfLayerPtr();
+  // Looping over the current submaps
+  for (const auto& id_submap_pair : id_to_submap_) {
+    // Getting the tsdf submap and its pose
+    const TsdfMap& tsdf_map = (id_submap_pair.second)->getTsdfMap();
+    const Transformation& T_M_S = (id_submap_pair.second)->getPose();
+    // Merging layers the submap into the global layer
+    mergeLayerAintoLayerB(tsdf_map.getTsdfLayer(), T_M_S,
+                          combined_tsdf_layer_ptr);
+  }
+  // Returning the new map
+  return projected_tsdf_map_ptr;
+}
+
+template <typename SubmapType>
+bool SubmapCollection<SubmapType>::setSubMapPose(const SubmapID submap_id,
+                                                 const Transformation& pose) {
+  // Looking for the submap
+  const auto tsdf_submap_ptr_it = id_to_submap_.find(submap_id);
+  if (tsdf_submap_ptr_it != id_to_submap_.end()) {
+    typename SubmapType::Ptr submap_ptr = (*tsdf_submap_ptr_it).second;
+    submap_ptr->setPose(pose);
+    return true;
+  } else {
+    std::cout << "Tried to set the pose of the submap with submap_id: "
```

Comment:
We should probably remove all std::cout calls and replace them with glog calls. As painful as that is.